### PR TITLE
terraform-providers.shell: uses default model

### PR DESCRIPTION
**Pull request by urbas**
This is a non-functional change.

###### Motivation for this change

Ever since Terraform providers added support for `buildGoModule` there's no longer any need for the `terraform-provider-shell` to be treated specially. 

This change removes the special treatment of the shell provider and converts it to the standard model.

The added benefit is that this provider can now be automatically updated via the `update-all-providers` script.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by urbas** [comment]
Result of `nixpkgs-review pr 112284` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>terraform-full</li>
    <li>terraform-providers.shell</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112284` at a9e03c40 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>terraform-full</li>
    <li><a href="https://gist.github.com/c9e7d2201f4a5dd23184e43783d186e3">terraform-providers.shell</a></li></li>
  </ul>
</details>
<details>
  <summary>4 warnings produced:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/cluster/terraform/default.nix:143:9:
```
    |
143 |         ./provider-path.patch
    |         ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/cluster/terraform/default.nix:144:9:
```
    |
144 |         (fetchpatch {
    |         ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/cluster/terraform/default.nix:155:17:
```
    |
155 |     patches = [ ./provider-path.patch ];
    |                 ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/cluster/terraform/default.nix:163:17:
```
    |
163 |     patches = [ ./provider-path.patch ];
    |                 ^
```
</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/terraform-providers/shell/default.nix [hunk 1]
```diff
@@ -1,32 +0,0 @@
...
-
-  vendorSha256 = "1p2ja6cw3dl7mx41svri6frjpgb9pxsrl7sq0rk1d3sviw0f88sg";
-
-  doCheck = false;
-
-  subPackages = [ "." ];
-
-  # Terraform allows checking the provider versions, but this breaks
-  # if the versions are not provided via file paths.
-  postInstall = "mv $out/bin/${pname}{,_v${version}}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/terraform-providers/shell/default.nix:21]: Is this lost?

Comment (urbas) [reply pkgs/applications/networking/cluster/terraform-providers/shell/default.nix:21]: It's not lost. It's done in [`teraform-providers/default.nix:L23`](https://github.com/NixOS/nixpkgs/blob/a9e03c40a1da210c17e3ea6f44a29e443c9d46ab/pkgs/applications/networking/cluster/terraform-providers/default.nix#L23).

The old `shell/default.nix` file replicates what's done in `default.nix`. Btw, the historical reason why `shell/default.nix` is in its own file is that it used `buildGoModule`, which was not supported by [the "standard model"](https://github.com/NixOS/nixpkgs/blob/a9e03c40a1da210c17e3ea6f44a29e443c9d46ab/pkgs/applications/networking/cluster/terraform-providers/default.nix#L43).

**Comment by kalbasit** [comment]
@GrahamcOfBorg build terraform.full

**Pull request merged**